cloud/azure/app-gateway/detectors-app-gateway.tf

```
@@ -0,0 +1,250 @@
+resource "signalfx_detector" "heartbeat" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Azure App Gateway heartbeat"
+
+  program_text = <<-EOF
+		from signalfx.detectors.not_reporting import not_reporting
+		signal = data('ResponseStatus', filter=filter('resource_type', 'Microsoft.Network/applicationGateways') and filter('primary_aggregation_type', 'true') and ${module.filter-tags.filter_custom}).publish('signal')
+		not_reporting.detector(stream=signal, resource_identifier=['HttpStatusGroup'], duration='${var.heartbeat_timeframe}').publish('CRIT')
+	EOF
+
+  rule {
+    description           = "has not reported in ${var.heartbeat_timeframe}"
+    severity              = "Critical"
+    detect_label          = "CRIT"
+    disabled              = coalesce(var.heartbeat_disabled, var.detectors_disabled)
+    notifications         = coalescelist(var.heartbeat_notifications, var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} on {{{dimensions}}}"
+  }
+}
+
+resource "signalfx_detector" "total_requests" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Azure App Gateway requests"
+
+  program_text = <<-EOF
+		signal = data('CurrentConnections', filter=filter('resource_type', 'Microsoft.Network/applicationGateways') and filter('primary_aggregation_type', 'true') and ${module.filter-tags.filter_custom})${var.total_requests_aggregation_function}.${var.total_requests_transformation_function}(over='${var.total_requests_transformation_window}').publish('signal')
```

Need a sum rollup here

```suggestion
		signal = data('CurrentConnections', filter=filter('resource_type', 'Microsoft.Network/applicationGateways') and filter('primary_aggregation_type', 'true') and ${module.filter-tags.filter_custom},  rollup='sum')${var.total_requests_aggregation_function}.${var.total_requests_transformation_function}(over='${var.total_requests_transformation_window}').publish('signal')
```

---

Ok, will update the script